### PR TITLE
Add generation instructions to core objects in preparation for Protobuf

### PR DESCRIPTION
**Pull request by smarterclayton**
All code changes are behind the 'proto' tag until generation is part of the build.

Also change ConvertToVersion to not SetVersionAndKind when version is internal (as our pattern).

Extracted from #16649

### CODE REVIEW

File: pkg/api/serialization_test.go [hunk 1]
```diff
@@ -246,54 +260,132 @@ func TestUnversionedTypes(t *testing.T) {
...
+	}
+	return items
+}
+
+// BenchmarkEncodeCodec measures the cost of performing a codec encode, which includes
+// reflection (to clear APIVersion and Kind)
+func BenchmarkEncodeCodec(b *testing.B) {
+	items := benchmarkItems()
+	width := len(items)
+	b.StartTimer()
```
Comment (lavalamp) [inline pkg/api/serialization_test.go]: I think you want to use ResetTimer, or call StopTimer first thing.

File: pkg/runtime/protobuf/protobuf.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+		return nil, err
+	}
+	return (&runtime.Unknown{
+		TypeMeta: runtime.TypeMeta{
+			Kind:       kind,
+			APIVersion: version,
+		},
+		RawJSON: b,
+	}).Marshal()
+}
```
Comment (lavalamp) [inline pkg/runtime/protobuf/protobuf.go]: tiny nit: add blank line

**Comment by lavalamp** [comment]
LGTM if you fix the benchmarks :)

**Comment by smarterclayton** [comment]
Fixes in place, applying tag

**Pull request merged**